sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSortExec.scala

```diff
@@ -178,6 +178,45 @@ case class GpuSortEachBatchIterator(
   }
 }
 
+/**
+ * Create an iterator that will sort each batch as it comes in. It will keep any projected
+ * columns in place after doing the sort on the assumption that you want to possibly combine
+ * them in some way afterwards.
+ */
+object GpuSpillableProjectedSortEachBatchIterator {
+  def apply(
+      iter: Iterator[ColumnarBatch],
+      sorter: GpuSorter,
+      opTime: GpuMetric = NoopMetric,
+      sortTime: GpuMetric = NoopMetric): Iterator[SpillableColumnarBatch] = {
+    val spillableIter = iter.flatMap { cb =>
+        // Filter out empty batches and make them spillable
+        if (cb.numRows() > 0) {
+          Some(SpillableColumnarBatch(cb, SpillPriorities.ACTIVE_ON_DECK_PRIORITY))
+        } else {
+          cb.close()
+          None
+        }
+    }
```

Comment:
If single batch is the common case it does not seem to matter but if we needed to save intermediate Option generation we could use an explicit PartialFunction instance:

```suggestion
    val spillableIter = iter.collect {
      // Filter out empty batches and make them spillable
      new PartialFunction[ColumnarBatch, SpillableColumnarBatch] {
        override def isDefinedAt(cb: ColumnarBatch): Boolean = if (cb.numRows() > 0) {
          true
        } else {
          cb.close()
          false
        }
        override def apply(cb: ColumnarBatch): SpillableColumnarBatch =
          SpillableColumnarBatch(cb, SpillPriorities.ACTIVE_ON_DECK_PRIORITY)
      }
    }
```

Reply:
I'll keep it in mind, but I'm not sure it matters that much here.